experiments/utils.py

```diff
@@ -366,8 +375,18 @@ def _get_disabled_experiment_names(self):
         )
 
     def set_disabled_experiments(self, names):
-        Enrollment.objects.filter(experiment__name__in=names).update(disabled=True)
-        Enrollment.objects.exclude(experiment__name__in=names).update(disabled=False)
+        for name in names:
+            experiment = experiment_manager.get_experiment(name, auto_create=False)
+            if not experiment:
+                continue
+            Enrollment.objects.update_or_create(
+                experiment=experiment,
+                user=self.user,
+                defaults={
+                    'disabled': True,
+                }
+            )
```

Comment:
There is a counter related to the enrollment of a participant to experiments, by enrolling people here, I say again the results will be influenced.

When I read the function named `set_disabled_experiments` and I see it enrolling users to experiments instead of **setting disabled experiments** I smell more bugs.

I think your work needs to be done in line [132](https://github.com/ConsumerAffairs/django-experiments/pull/22/files#diff-fbac30eee3a7d2644ac17c552e00a34aR132) in a way you get the most up-to-date list of all of the disabled experiments at that point.

Reply:
Please demonstrate the problem, I haven't been able to find any.

```
            Enrollment.objects.update_or_create(
                experiment=experiment,
                user=self.user,
                defaults={
                    'disabled': True,
                }
            )
```
This does not actually enroll anyone, because the variant is set to `None`.

Let's have a screenshare please.

Reply:
I haven't looked at the source code if `django-experiments` yet, unfortunately. But it feels really weird to `_or_create` objects in a method called `set_disabled_experiments`. Sounds like the function is doing more than it should? 🤔

Reply:
That’s Ivan’s concern as well. One solution would be to keep the `disabled` flag on another (new) model.